Source/HttpMultipartParser/MultipartFormDataParser.cs

```diff
@@ -557,10 +557,11 @@ private void ParseStream(Stream stream, string boundary, Encoding encoding, int
             var streamingParser = new StreamingMultipartFormDataParser(stream, boundary, encoding, binaryBufferSize);
             streamingParser.ParameterHandler += parameterPart => Parameters.Add(parameterPart);
 
-            streamingParser.FileHandler += (name, fileName, type, disposition, buffer, bytes) =>
+            streamingParser.FileHandler += (name, fileName, type, disposition, buffer, bytes, partNo) =>
             {
-                if (Files.Count == 0 || name != Files[Files.Count - 1].Name)
+                if (partNo == 0)

```

Comment:
I would rename in order to be consistent

```suggestion
                if (partNumber == 0)
```